### PR TITLE
Social Media Manager

### PR DESCRIPTION
**Pull request by WellnessSupplyCo**
This plugin allows you to manage your social media accounts directly from Auto-GPT, freeing up your time for more important tasks. With support for Facebook, Instagram, Twitter, and Google My Business, you can automate your social media tasks and focus on what really matters.

### CODE REVIEW

File: src/autogpt_plugins/SocialMediaManager/__init__.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+        Args:
+            prompt (PromptGenerator): The prompt generator.
+
+        Returns:
+            PromptGenerator: The prompt generator.
+        """
+	# Instagram commands
+        if connect_ig():
+            prompt.add_command("load_ig_credentials", "Load Instagram Credentials", {}, load_ig_credentials)
+            prompt.add_command("connect_ig", "Connect Instagram", {}, connect_ig)
```
Comment (ntindle) [inline src/autogpt_plugins/SocialMediaManager/__init__.py:317]: Don't let the ai do this

File: src/autogpt_plugins/SocialMediaManager/__init__.py [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+            prompt.add_command("ig_get_direct_messages", "Get Instagram Direct Messages", {"thread_id": "<thread_id>"}, ig_get_direct_messages)
+            prompt.add_command("ig_get_hashtag_posts", "Get Instagram Hashtag Posts", {"hashtag": "<hashtag>"}, ig_get_hashtag_posts)
+            prompt.add_command("ig_get_user_stories", "Get Instagram User Stories", {"user_id": "<user_id>"}, ig_get_user_stories)
+            prompt.add_command("fb_get_ig_post_insights", "Get Instagram Post Insights", {"post_id": "<post_id>"}, fb_get_ig_post_insights)
+
+
+	# Facebook commands
+        if connect_fb():
+            prompt.add_command("load_fb_credentials", "Load Facebook Credentials", {}, load_fb_credentials)
+            prompt.add_command("connect_fb", "Connect Facebook", {}, connect_fb)
```
Comment (ntindle) [inline src/autogpt_plugins/SocialMediaManager/__init__.py:336]: Don't let the ai do this

File: src/autogpt_plugins/SocialMediaManager/__init__.py [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+            prompt.add_command("fb_get_page_events", "Get Facebook Page Events", {"page_id": "<page_id>"}, fb_get_page_events)
+            prompt.add_command("fb_create_page_event", "Create Facebook Page Event", {"page_id": "<page_id>", "event_data": "<event_data>"}, fb_create_page_event)
+            prompt.add_command("fb_get_group_members", "Get Facebook Group Members", {"group_id": "<group_id>"}, fb_get_group_members)
+            prompt.add_command("fb_get_group_posts", "Get Facebook Group Posts", {"group_id": "<group_id>"}, fb_get_group_posts)
+
+
+	# Twitter commands
+        if connect_tw():
+            prompt.add_command("tw_post_tweet", "Post Tweet", {"tweet_text": "<tweet_text>"}, tw_post_tweet)
+            prompt.add_command("tw_post_media_tweet", "Post Media Tweet", {"tweet_text": "<tweet_text>", "media_path": "<media_path>"}, tw_post_media_tweet)
```
Comment (ntindle) [inline src/autogpt_plugins/SocialMediaManager/__init__.py:355]: Don't let the ai do this

File: src/autogpt_plugins/SocialMediaManager/__init__.py [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+        
+        load_ig_credentials()
+        load_fb_credentials()
+        load_tw_credentials()
+        load_gmb_credentials()
+
+        connect_ig()
+        connect_fb()
+        connect_tw()
+        connect_gmb()
```
Comment (ntindle) [inline src/autogpt_plugins/SocialMediaManager/__init__.py:33]: make all of these Social Media platforms classes so they aren't polluting the global context (and later on we can run multiple social accounts by loading the plugin with different configs)

**Review by ntindle — CHANGES_REQUESTED** [review]
Few comments, 

Needs tests

File: src/autogpt_plugins/SocialMediaManager/__init__.py (hunk 4 above)
Comment (WellnessSupplyCo) [reply src/autogpt_plugins/SocialMediaManager/__init__.py:33]: Appreciate it! Still learning. Will have an update later after work.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team to all open Plugin-PRs

We're in the process of re-architecting:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

Please be patient as we work to clean up the AutoGPT core.

If you already have an existing plug-in that’s affected, please message @ntindle on Discord so we can get your plugin spinning on the new 
architecture.

If you're working on a plugin, please bear in mind that one of our primary goals in re-architecting is to make life MUCH easier for you. 
We've taken aboard feedback from plugin developers. We're on it, so please give us some time!

<3 The AutoGPT Team

**Pull request closed without merging**